ebean-test/src/test/java/org/tests/basic/TestLoadBeanCache.java

```diff
@@ -35,14 +35,15 @@ void testLoad() {
     Map<String, Country> map = DB.find(Country.class)
       .setBeanCacheMode(CacheMode.PUT)
       .setUseQueryCache(true)
-      .setReadOnly(true)
+      //.setReadOnly(true)
+      .setUnmodifiable(true)
```

Comment:
This shouldn't be neccessary (see other comment) `setUseQueryCache(true)` should be enough.

Reply:
> We have exactly one (theoretical) use case, where we modify a bean [that has come out of the query cache]

The query cache results are always unmodifiable. So say we have a use case where we've got an unmodifiable bean (from either a query cache result or just a query.setUnmodifiable()) and we need to perform some mutation on it and perform an update. How can we do it? 

Options:
- 1: Manually create a mutable copy of the bean, mutate, then database.update() it
- 2: Add some **_new API_** that creates a mutable copy of an entity bean like `<T> T database.mutableCopy(T source);`

Interestingly enough, a mutable copy instance could actually allow lazy loading (makes me think of the lazy loading lob case mentioned above). Hmmm.